frontend/src/metabase/components/Modal/WindowModal.tsx

```diff
@@ -55,12 +57,15 @@ export class WindowModal extends Component<WindowModalProps> {
       this._modalElement.style.zIndex = String(props.zIndex);
     }
 
-    const modalContainer =
-      window.document.querySelector(
-        `#${EMBEDDING_SDK_PORTAL_ROOT_ELEMENT_ID}`,
-      ) || document.body;
+    if (props.isOpen) {
+      getPortalRootElement().appendChild(this._modalElement);
+    }
+  }
 
-    modalContainer.appendChild(this._modalElement);
+  componentDidUpdate(prevProps: WindowModalProps) {
+    if (!prevProps.isOpen && this.props.isOpen) {
+      getPortalRootElement().appendChild(this._modalElement);
+    }
```

Comment:
Do we also need to cleanup the appended modal element from the DOM when the window modal closes?

Reply:
Yes. I think that happens in the `componentWillUnmount` below, but if I missed something, let me know